amazonfreertossdk/src/main/java/software/amazon/freertos/amazonfreertossdk/AmazonFreeRTOSDevice.java

```diff
@@ -478,9 +470,13 @@ public void onMtuChanged(BluetoothGatt gatt, int mtu, int status)  {
                     mMtu = mtu;
                     mMaxPayloadLen = mMtu - 3;
                     mMaxPayloadLen = mMaxPayloadLen > 0 ? mMaxPayloadLen : 0;
+                    // The BLE service should be initialized at this stage
+                    if ( mBleConnectionState == BleConnectionState.BLE_INITIALIZING) {
```

Comment:
As  a side effect of this change `sendConnAck()`  at line 756 is failing since it checks for `BLE_CONNECTED` state before sending.Suggesting to move the check for BLE state inside `sendConnAck()` definition and check for `isBleConnected()` instead, as we do for other MQTT relay methods.